### PR TITLE
[BUGFIX] Fix remaining FDS config substitution issues

### PR DESCRIPTION
**Pull request by Kilo59**
- Followup to https://github.com/great-expectations/great_expectations/pull/7914

Fixes remaining issues with S3 and GCS datasource config substitution.
Fixes an early return as part of the recursive `ConfigStr` substitutions.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Comment by ghost** [comment]
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=7917&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=7917&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map legend
</details>

File: great_expectations/datasource/fluent/config_str.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def get_config_value(self, config_provider: _ConfigurationProvider) -> str:
         Resolve the config template string to its string value according to the passed
         _ConfigurationProvider.
         """
+        LOGGER.info(f"Substituting '{self}'")
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/config_str.py:42]: Note this logs the config template string, not the potentially sensitive substituted value.

File: great_expectations/datasource/fluent/fluent_base_model.py [hunk 2]
```diff
@@ -288,19 +288,19 @@ def __str__(self):
...
             elif isinstance(v, (MutableMapping, MutableSequence)):
-                return _recursively_set_config_value(v, config_provider)
+                _recursively_set_config_value(v, config_provider)
     elif isinstance(data, MutableSequence):
         for i, v in enumerate(data):
             if isinstance(v, ConfigStr):
                 data[i] = v.get_config_value(config_provider)
             elif isinstance(v, (MutableMapping, MutableSequence)):
-                return _recursively_set_config_value(v, config_provider)
+                _recursively_set_config_value(v, config_provider)
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/fluent_base_model.py:303]: This was a boneheaded move by me.
The return was causing an early exit.

File: great_expectations/datasource/fluent/config.py [hunk 3]
```diff
@@ -222,7 +222,7 @@ def _load_datasource_subtype(cls, v: List[dict]):
                 for asset in datasource.assets:
                     asset._datasource = datasource
 
-        logger.info(f"Loaded 'datasources' ->\n{repr(loaded_datasources)}")
+        logger.debug(f"Loaded 'datasources' ->\n{repr(loaded_datasources)}")
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/config.py:225]: This is way too verbose in our test logs when logging more than 1 or 2 datasources.

**Review by alexsherstinsky — APPROVED** [review]
LGTM -- this is great!  Thank you!

**Pull request merged**